src/modules/ClientApi.ts

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2025 Element Creations Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+import type { ClientApi as IClientApi, Room } from "@element-hq/element-web-module-api";
+import { Room as ModuleRoom } from "./models/Room";
+import { AccountDataApi } from "./AccountDataApi";
+import { MatrixClientPeg } from "../MatrixClientPeg";
+
+export class ClientApi implements IClientApi {
+    private accountDataApi?: AccountDataApi;
+
+    public getRoom(roomId: string): Room | null {
+        const sdkRoom = MatrixClientPeg.safeGet().getRoom(roomId);
+        if (sdkRoom) return new ModuleRoom(sdkRoom);
+        return null;
+    }
+
+    public get accountData(): AccountDataApi {
+        this.accountDataApi ??= new AccountDataApi();
+        return this.accountDataApi;
+    }
```

Comment:
couldn't this just be `public readonly accountData = new AccountDataApi();` in the class preamble?

Reply:
https://github.com/element-hq/element-web/pull/31138/commits/5f8aa3201539ecd923713b413198f07307ba643b